parsl/data_provider/ftp.py

```diff
@@ -50,15 +50,20 @@ def replace_task(self, dm, executor, file, f):
         return in_task_transfer_wrapper(f, file, working_dir)
 
 
-def in_task_transfer_wrapper(func, file, working_dir):
+def in_task_transfer_wrapper(func, file, working_dir, secure_ftp=False):
```

Comment:
This new parameter, and the corresponding one added to _ftp_stage_in, are not exposed to the end user anywhere - so I think this new code path won't ever be taken?

Reply:
I've added the FTP_TLS functionality in case users ever want to use the `secure_ftp` mode. The method is called by `replace_task`. If anyone's interested to use the secure mode, they can set the parameter value to `True`.

Reply:
Right, the method is called by `replace_task` without setting that parameter either way -- just using the default. So there isn't any user expressible workflow/configuration that will set this parameter to true.

Reply:
Then, I should probably add parameter `secure_ftp` to `replace_task` method which will call the `in_task_transfer_wrapper` method using the parameter. Sounds okay?

Reply:
I think you could probably expose this as an option for users by giving FTPSeparateTaskStaging and FTPInTaskStaging init methods to make them configurable, like the GlobusStaging provider (or like other configurable Parsl components in general).

Reply:
Thanks for the idea! Sounds great!

Reply:
If you store it into `self.secure_ftp` in `__init__` you'll be able to pass it into the two functions you already changed, without having to pass it via replace_task.

Reply:
I was able to change it for `FTPInTaskStaging` but not sure how I'd do it for `FTPSeparateTaskStaging` and `_ftp_stage_in`. Can you please help?

Reply:
I don't have time to help with this right now - ping me in exactly 48 hours and I should have a bit more time.

Reply:
@benclifford  Pinging you according to your last message.

Reply:
ok so where are you getting stuck? what code did you try so far?

Reply:
I'm stuck on class "FTPSeparateTaskStaging" since it calls the ‘_ftp_stage_in_app‘ method which calls ‘_ftp_stage_in‘ method by passing it.

Not sure how I'd pass the parameter 'secure_ftp'.

Reply:
have a look at how the `working_dir` parameter gets passed in

Reply:
Thanks! It should be good to go now.